### ISSUE TITLE
psnr

### ISSUE THREAD

**Issue by zhaoyucong** [tegg89/SRCNN-Tensorflow#51]
I can't measure the PSNR in the paper with this code, and the PSNR in Bicubic is also different from that in the paper. Can you give me some comments@tegg89

**Comment by IronLavender** [comment]
Is there anyone who get the psnr result. The calculation is for a whole image instead of sub_image? How to calculate the PSNR when images and label aren't in same shape?

**Comment by manmeet3591** [comment]
> I can't measure the PSNR in the paper with this code, and the PSNR in Bicubic is also different from that in the paper. Can you give me some comments@tegg89

@zhaoyucong I tried the function below for calculating psnr, although visually I can see that srcnn is better, but psnr values are lower for srcnn as compared to bicubic

def psnr(img1, img2):
    mse = numpy.mean( (img1 - img2) ** 2 )
    if mse == 0:
        return 100
    PIXEL_MAX = 255.0
    return 20 * math.log10(PIXEL_MAX / math.sqrt(mse))



> Is there anyone who get the psnr result. The calculation is for a whole image instead of sub_image? How to calculate the PSNR when images and label aren't in same shape?

@IronLavender If you carefully check the image and label, you can basically see that if you crop the image, with padding as defined, you get one to one comparison with label

**Comment by jb2020-super** [comment]
The key point is that you should use the _imresize_ of MatLab as the paper said both for training and testing. Good luck!

**Comment by WangberlinT** [comment]
if you use train_data(bicubic patches sequence nx31x31) and train_label(ground truth sequence: nx21x21) calculate PSNR, first you should make them have the same size, second you should calculate as 20 * math.log10(1.0 / math.sqrt(mse)), because they have been normalized already.
